nemo_skills/inference/server/reward_model.py

```diff
@@ -106,14 +106,31 @@ def __init__(self, **kwargs):
         self.model = model_list.data[0].id
 
     def score(self, prompts: list[str]) -> list[float]:
-        # TODO: The current VLLM support for Qwen-RM uses a hack of using embedding APIs.
-        # Once VLLM officially adds the support, change the API.
-        responses = self.oai_client.embeddings.create(input=prompts, model=self.model)
-        outputs = []
-        for data in responses.data:
-            raw_score = data.embedding[-1]
+        outputs = [None] * len(prompts)  # Pre-allocate a list to store results in correct order
+        futures = {}
+
+        def process_single_prompt(prompt):
```

Comment:
Can you make this a separate method named "_score_single_seq" 
- Reason for a separate method: Easier to read. Typically a function within a function makes sense when the inner function needs some local variables from the outer function. Since here there is no such tight coupling, we can use a separate method.
- Reason for the name change: The "_" prefix is to denote it is for internal use, "score" is more specific than "process", and "seq" is more apt than "prompt" because we are feeding in the sequence of Input + Output. 

May be good to rename the local var `prompt/prompts` to `seq/seqs`

Reply:
Ok, will revise on it.

Reply:
I checked other parts include Nemo-reward model, we all use prompt instead of seq.
So it is better to use prompt as the variable for consistency or we have to revise other parts as well.